tests/test_nhg.py

```diff
@@ -216,6 +287,358 @@ def test_route_nhg(self, dvs, dvs_route, testlog):
+
+        def get_nhgm_keys():
+            return asic_db.get_keys(self.ASIC_NHGM_STR)
+
+        def get_rt_keys():
+            return asic_db.get_keys(self.ASIC_RT_STR)
+
+        # Count existing objects
+        prev_nhg_keys = get_nhg_keys()
+        asic_nhgs_count = len(get_nhg_keys())
+        asic_nhgms_count = len(get_nhgm_keys())
+        asic_nhs_count = len(asic_db.get_keys(self.ASIC_NHS_STR))
+
+        # Remove a group that does not exist
+        nhg_ps._del("group1")
+        time.sleep(1)
+        assert len(get_nhg_keys()) == asic_nhgs_count
+
+        # Create a next hop group with a member that does not exist - should fail
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3,10.0.0.63'),
+                                        ("ifname", "Ethernet0,Ethernet4,Ethernet124")])
+        nhg_ps.set("group1", fvs)
+        time.sleep(1)
+        assert len(asic_db.get_keys(self.ASIC_NHG_STR)) == asic_nhgs_count
+
+        # Issue an update for this next hop group that doesn't yet exist,
+        # which contains only valid NHs.  This will overwrite the previous
+        # operation and create the group.
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.5'),
+                                        ("ifname", "Ethernet0,Ethernet8")])
+        nhg_ps.set("group1", fvs)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 1)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+
+        # Check the group has it's two members
+        for nhgid in asic_db.get_keys(self.ASIC_NHG_STR):
+            if nhgid not in prev_nhg_keys:
+                break
+
+        count = 0
+        for k in asic_db.get_keys(self.ASIC_NHGM_STR):
+            fvs = asic_db.get_entry(self.ASIC_NHGM_STR, k)
+            if fvs['SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID'] == nhgid:
+                count += 1
+        assert count == 2
+
+        # Add a route referencing the new group
+        asic_rts_count = len(get_rt_keys())
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', 'group1')])
+        rt_ps.set('2.2.2.0/24', fvs)
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_rts_count + 1)
+
+        # Get the route key
+        for rt_key in get_rt_keys():
+            k = json.loads(rt_key)
+
+            if k['dest'] == "2.2.2.0/24":
+                break
+
+        # Try removing the group while it still has references - should fail
+        nhg_ps._del('group1')
+        time.sleep(1)
+        assert len(get_nhg_keys()) == asic_nhgs_count + 1
+
+        # Create a new group
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('ifname', 'Ethernet0,Ethernet4')])
+        nhg_ps.set("group2", fvs)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 2)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 4)
+
```

Comment:
also check for "asic_nhs_count == 3??"

Reply:
I think that's redundant. As I see it, `test_nhg.py` should test the NHG related code, assuming the existing code does it's job just fine. In this scenario, we should assume that the neighOrch will create those NHs fine, because it has been tested on its own as well. Even so, there will be an underlying check that the expected NHs do in fact exists as the NHGs couldn't be created otherwise.